src/odemis/gui/model/__init__.py

```diff
@@ -595,6 +595,40 @@ def __init__(self, main):
         # VA for autofocus procedure mode
         self.autofocus_active = BooleanVA(False)
 
+
+class LocalizationGUIData(MicroscopyGUIData):
+    # TODO: update the doc string
+    """ Represent an interface used to only show the current data from the microscope.
+
+    It should be able to handle SEM-only, optical-only, SECOM and DELPHI systems.
+
+    """
+
+    def __init__(self, main):
+        assert main.microscope is not None
```

Comment:
Maybe it is nice to add a message if the assertion fails so that the user of the class knows what went wrong. Something like:
> assert main.microscope is not None, "Class .... can only be instantiated with the main.microscope is defined"

Reply:
To be fair to Mahmoud, this is just copy-paste from the other similar classes.

The reason it's like this on the other classes is that I'm personally not a fan of adding such a message to `assert`. I feel that if you really want to care about some states, then you should use a proper if + Exception. I like to use `assert` as a very short reminder for the reader about the code's expectations.

Reply:
Do you mean if + raise? 
What kind of error should be raised in the situation of missing/unconnected component?

Reply:
Yes, if + raise ValueError().